src/Pate/Verification.hs

```diff
@@ -545,8 +545,8 @@ withPair pPair f = do
 provePostcondition ::
```

Comment:
What is the meaning of this `DomainSpec`? What is it meant to be used for?

I think that the generated proof term makes sense

Reply:
Added some documentation.